src/v/security/gssapi_authenticator.cc

```diff
@@ -111,55 +112,126 @@ void display_status(
     display_status_1(msg, min_stat, GSS_C_MECH_CODE);
 }
 
+// impl runs on the thread_worker
+class gssapi_authenticator::impl {
+public:
+    using state = gssapi_authenticator::state;
+
+    template<typename R>
+    struct state_result {
+        gssapi_authenticator::state state;
+        result<R> result;
+    };
+
+    impl(
+      ss::sstring primary, ss::sstring keytab, std::vector<gssapi_rule> rules)
+      : _primary{std::move(primary)}
+      , _keytab{std::move(keytab)}
+      , _rules{std::move(rules)} {}
+
+    state_result<bytes> authenticate(bytes auth_bytes);
+    const security::acl_principal& principal() const { return _principal; }
+
+private:
+    state_result<void> init();
+    state_result<bytes> more(bytes_view);
+    state_result<bytes> ssfcap(bytes_view);
+    state_result<bytes> ssfreq(bytes_view);
+    state_result<void> check();
+    void
+    fail_impl(OM_uint32 maj_stat, OM_uint32 min_stat, std::string_view msg);
+    template<typename... Args>
+    void fail(
+      OM_uint32 maj_stat,
+      OM_uint32 min_stat,
+      fmt::format_string<Args...> format_str,
+      Args&&... args) {
+        auto msg = ssx::sformat(format_str, std::forward<Args>(args)...);
+        fail_impl(maj_stat, min_stat, msg);
+    }
+    acl_principal get_principal_from_name(std::string_view source_name);
+
+    ss::sstring _primary;
```

Comment:
Should this be called out specifically as the _kerberos_principal?  The primary is only part of the overall principal.

Reply:
> Should this be called out specifically as the _kerberos_principal? The primary is only part of the overall principal.

It's the primary of the Kerberos SPN. It's a bit close to `_principal`, I agree.

How about:
* `_krb_service_primary`
* `_rp_user_principal`

Reply:
For the redpanda won't this value be "redpanda/<host>@REALM"?  Or is it just "redpanda"?

Reply:
I'll refer you to the docs: https://github.com/BenPope/redpanda/blob/2beefd11c924456754d759c43453bdaf93eb62f9/src/v/config/configuration.cc#L899

It's a cluster config property, so it has to be the same for each broker in the cluster. 

The hostname is resolved here: https://github.com/BenPope/redpanda/blob/2beefd11c924456754d759c43453bdaf93eb62f9/src/v/security/gssapi_authenticator.cc#L237-L238

Reply:
thanks sorry

Reply:
> thanks sorry

No apology needed!